test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/DacFx/DacFxServiceTests.cs

```diff
@@ -868,6 +869,35 @@ public void ValidateObjectTypesOptionswithEnum()
             }
         }
 
+        // <summary>
+        /// Verify that publish profile gets created and saved
+        /// </summary>
+        [Test]
+        public async Task ValidateSavePublishProfile()
+        {
+            DacFxService service = new DacFxService();
+            string folderPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "DacFxTest");
+            string profileFilePath = Path.Combine(folderPath, string.Format("validateSavePublishProfile.publish.xml"));
+
+            var savePublishProfileParams = new SavePublishProfileParams
+            {
+                ProfilePath = profileFilePath,
+                DatabaseName = "testDb",
+                ConnectionString = "testConnString",
+                SqlCommandVariableValues = new Dictionary<string, string>()
+                    {
+                        { "testvar", "testval" }
+                    }
+            };
+
+            var dacfxRequestContext = new Mock<RequestContext<ResultStatus>>();
+            dacfxRequestContext.Setup((RequestContext<ResultStatus> x) => x.SendResult(It.Is<ResultStatus>((result) => result.Success == true))).Returns(Task.FromResult(new object()));
```

Comment:
You could use the new MockRequest helpers to clean this up a bit: https://github.com/microsoft/sqltoolsservice/blob/main/test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SqlProjects/SqlProjectsServiceTests.cs#L34